eox_core/api/v1/serializers.py

```diff
@@ -196,14 +198,47 @@ def update(self, instance, validated_data):
         """
         Update method for safe fields.
         """
+        # Obtain only the User profile fields defined in the EdxappExtendedUserSerializer
+        all_fields = set(EdxappExtendedUserSerializer().fields)
+        non_profile_fields = set(EdxappUserSerializer().fields)
+        extra_registration_fields = all_fields - non_profile_fields
+        # Check if the User has a Profile
+        has_profile = hasattr(instance, 'profile')
+
+        extended_profile_fields = getattr(settings, 'extended_profile_fields', [])
+        extra_registration_fields.update(extended_profile_fields)
+        if has_profile:
+            profile_meta = instance.profile.get_meta()
```

Comment:
why do you get profile_meta here and in line 220 you empty it?

Reply:
@mariajgrimaldi thank you for noticing, I forgot to add the condition `if not has_profile` to ensure that this only happens if the user currently does not have a UserProfile and a profile field wants to be added